develop/tutorials/articles/470-javascript/05-using-single-page-applications.markdown

```diff
@@ -0,0 +1,289 @@
+
+Now that you have a better understanding of how SennaJS benefits SPA, you can 
+learn how to configure options for SPA within your @product@ installation next.
+
+## Customizing SPA Settings
+
+As stated earlier, SPA is enabled by default in your @product@ installation and
+applications. The default SPA configuration may suit your needs, however, there 
+may be certain use cases where a custom configuration is more desirable.
+
+Depending on what behaviors you need to customize, you can configure SPA options 
+in one of two places. SPA caching and SPA timeout settings can be configured in 
+System Settings. If you wish to disable SPA for a certain link, page, or portlet
+in your site, you can do so within the corresponding element itself. All SPA 
+configuration options are covered in this section.
+
+### Configuring SPA System Settings
+
+To configure system settings for SPA, follow these steps:
+
+1. Navigate to *Configuration* &rarr; *System Settings* within the Control
+   Panel.
+
+2. Select the *Foundation* tab at the top of the page.
+
+3. Click *Frontend SPA Infrastructure*.
+
++$$$
+
+**Note:** In prior versions of Liferay, all SPA render requests that didn't 
+belong to a portlet (no `p_p_id` in the URL) were cached indefinitely. This can 
+confuse users, as the content they view will be cached, rather than the latest 
+fresh content. Since Liferay Portal CE 7.0 GA2 and Liferay DXP 7.0 GA1, 
+administrators can use the **Cache Expiration Time** property to set an 
+expiration time for the Senna cache.
+
+$$$
+
+The following configuration options are available:
+
+**Cache Expiration Time**:  The time, in minutes, in which the SPA cache is 
+cleared. A zero value means the cache should never expire during SPA navigation. 
+A negative value means the cache should be disabled.
+
+**Request Timeout Time**: The time, in milliseconds, in which a SPA request 
+times out. A zero value means the request should never timeout.
+
+**User Notification Time**: The time, in milliseconds, in which a notification 
+is shown to the user stating that the request is taking longer than expected. A 
+zero value means no notification should be shown.
+
+Now that you know how to configure system settings for SPA, you can learn how to
+disable SPA for elements in your site next.
+
+### Disabling SPA
+
+Certain elements of your page may require a page refresh to work properly. For 
+example, you may have downloadable content that you want to share with the user. 
+In these cases, SPA must be disabled for those specific elements.
+
+If there is a certain portlet or page that you don't want to be part of the SPA, 
+and instead require a full page navigation, you have some options:
+
+-  Blacklist the portlet to disable SPA for the entire portlet
+-  Use the `data-senna-off` annotation to disable SPA for a specific link
+
+<!-- How do you disable SPA for a page? Since it's mentioned as an option, we need to demonstrate it. -->
```

Comment:
How do you disable SPA for a page? Since it's mentioned as an option, we need to demonstrate it. This only shows how to disable SPA for a portlet or a link.
thx!